lldb/unittests/Core/MangledTest.cpp

```diff
@@ -435,142 +435,202 @@ struct DemanglingPartsTestCase {
 DemanglingPartsTestCase g_demangling_parts_test_cases[] = {
     // clang-format off
    { "_ZNVKO3BarIN2ns3QuxIiEEE1CIPFi3FooIS_IiES6_EEE6methodIS6_EENS5_IT_SC_E5InnerIiEESD_SD_",
-     { /*.BasenameRange=*/{92, 98}, /*.ScopeRange=*/{36, 92}, /*.ArgumentsRange=*/{ 108, 158 },
-       /*.QualifiersRange=*/{158, 176}, /*.PrefixRange=*/{0, 0}, /*.SuffixRange=*/{0, 0} },
+     {
+       /*.BasenameRange=*/{92, 98}, /*.TemplateRange=*/{0, 0}, /*.ScopeRange=*/{36, 92},
```

Comment:
I'm concerned how setting the `TemplateRange` is done outside of the `TrackingOutputBuffer`. Would it be difficult to track this inside the demangler?

Also, shouldn't the `CPlusPlusLanguage` plugin be changed to now use the `TemplateArgumentsRange` to handle the `${function.template-arguments}` variable?